packages/cover-traffic-daemon/Dockerfile

```diff
@@ -1,56 +1,69 @@
-# Run hopr-cover-traffic-daemon within a single container using npm
+ARG HOPR_TOOLCHAIN_IMAGE=${HOPR_TOOLCHAIN_IMAGE:-hopr-toolchain:latest}
+
+# Creates a Docker container that encapsulates a cover-traffic node
 
 # use slim version of node on Debian for smaller image sizes
-FROM node:16-bullseye-slim@sha256:95d4fdef5e12916a89f7d56dca6d8099de61df875c3c61106cb53e0ba1a48d44 as build
-
-# python is used by some nodejs dependencies as an installation requirement
-RUN apt-get update && \
-    apt-get install -y \
-    git \
-    python3 \
-    build-essential
-
-# enable to pass the version to Docker using either --build-arg or an
-# environment variable
-# if its not given, yarn will install the latest version of the package
-ARG PACKAGE_VERSION
-RUN echo "ARG PACKAGE_VERSION=${PACKAGE_VERSION}"
-ENV PACKAGE_VERSION=${PACKAGE_VERSION:-}
-RUN echo "ENV PACKAGE_VERSION=${PACKAGE_VERSION}"
+FROM ${HOPR_TOOLCHAIN_IMAGE} as build
+
+WORKDIR /app/hoprnet
+
+COPY .yarn .yarn
+COPY Makefile Cargo.toml Cargo.lock yarn.lock tsconfig.build.json tsconfig.json tsconfig.scripts.json yarn.lock rust-toolchain.toml ./
+COPY packages/ ./packages
+
+LABEL description="Launches a cover-traffic node which keeps sending cover-traffic through the hopr network"
 
 # making sure some standard environment variables are set for production use
 ENV NODE_ENV production
 ENV NODE_OPTIONS=--max_old_space_size=4096
 ENV npm_config_build_from_source false
 
-WORKDIR /app
+RUN CI=true make deps package=hopr-cover-traffic-daemon
 
-# install hoprd as a local module
-RUN yarn add @hoprnet/hopr-cover-traffic-daemon@${PACKAGE_VERSION}
+RUN NO_NEXT=true make -j build package=cover-traffic-daemon
 
-# use slim version of node on Debian for smaller image sizes
-FROM node:16-bullseye-slim@sha256:95d4fdef5e12916a89f7d56dca6d8099de61df875c3c61106cb53e0ba1a48d44 as runtime
+# Remove node_modules as they contain devDependencies
+RUN rm -R node_modules
+
+# No need for hoprd or AVADO
+RUN rm -R packages/hoprd packages/avado
+
+# Rebuild node_modules directory without devDependencies
+RUN CI=true NO_CARGO=true PRODUCTION=true make deps package=hopr-cover-traffic-daemon
+
+# Remove all typescript files
+RUN find ./packages -type f -name '*.ts' -delete
+
+FROM alpine:16@sha256:1304f174557314a7ed9eddb4eab12fed12cb0cd9809e4c28f29af86979a3c870 as runtime
```

Comment:
It would be good to know this hash to which version belongs in a comment

Reply:
changed